### PR TITLE
#115146851 A user should be able to receive notifications on reactions to their posts via email

### PR DESCRIPTION
**Pull request by NdagiStanley**
#### What does this PR do?
Allows the user to set whether they want notifications of likes and/or comments to their resources sent to their emails
#### Description of Task to be completed?
Add checkbox options to the `settings` page for specifying the preference
#### How should this be manually tested?
If you checked-on the checkbox for like and/or comment, you are able to receive an email notification
#### What are the relevant pivotal tracker stories?
#115146851
#### Screenshots
screenshot 2016-06-07 11 50 44

### CODE REVIEW

File: codango/account/templates/account/layout/base.html [hunk 1]
```diff
@@ -85,11 +85,23 @@
     {% block script %}{% endblock script %}
     <script src="{% static 'lib/bootstrap/dist/js/bootstrap.min.js' %}"></script>
     <script src='https://cdn.firebase.com/js/client/2.2.1/firebase.js'></script>
+    <script type="text/javascript" src="{% static 'lib/js-cookie/src/js.cookie.js' %}"></script>
     <script type="text/javascript" src="{% static 'js/app.js' %}"></script>
     <script type="text/javascript" src="{% static 'js/snippet.js' %}"></script>
     <script type="text/javascript" src="{% static 'js/app.react.js' %}"></script>
     <script type="text/javascript" src="{% static 'endless_pagination/js/endless-pagination.js' %}"></script>
     <script src="{% static 'lib/prettify-min/prettify.min.js' %}"></script>
+    <script type="text/javascript">
```
Comment (andela-cnnadi) [inline codango/account/templates/account/layout/base.html]: Can you please get rid of this commented out code.

File: codango/account/helper.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                'author': author,
+                'resource_link': request.build_absolute_uri(
+                    resource_link),
+                'settings_link': request.build_absolute_uri(
+                    '/user/' + author.username + '/settings'
+                )
+            }
+            send_notification.apply_async(
+                kwargs=data,
+                countdown=int(os.getenv('CELERY_NOTIFICATION_TIMEOUT')),
```
Comment (andela-cnnadi) [inline codango/account/helper.py]: Let's try to put this on a new line. This way we can do something like this:

```py
# Sets default timeout to 10 seconds if env variable is not found
celery_timeout = os.getenv('CELERY_NOTIFICATION_TIMEOUT') or 600
```

File: codango/account/templates/account/layout/base.html [hunk 3]
```diff
@@ -84,12 +84,24 @@
...
     <script src='https://cdn.firebase.com/js/client/2.2.1/firebase.js'></script>
+    <script type="text/javascript" src="{% static 'lib/js-cookie/src/js.cookie.js' %}"></script>
     <script type="text/javascript" src="{% static 'js/app.js' %}"></script>
     <script type="text/javascript" src="{% static 'js/snippet.js' %}"></script>
     <!-- React bundled js -->
     <script type="text/javascript" src="{% static 'js/build/bundle.js' %}"></script>
     <script type="text/javascript" src="{% static 'endless_pagination/js/endless-pagination.js' %}"></script>
     <script src="{% static 'lib/prettify-min/prettify.min.js' %}"></script>
+    <script type="text/javascript">
+        // $(window).on('beforeunload', function(e){
```
Comment (andela-cnnadi) [inline codango/account/templates/account/layout/base.html]: *Nitpick*: You need to get rid of this commented out code.

File: codango/account/urls.py [hunk 4]
```diff
@@ -9,7 +9,7 @@
         views.SearchView.as_view(), name='search_by'),
     url(r'^login$', views.LoginView.as_view(), name='login'),
     url(r'^register$', views.RegisterView.as_view(), name='register'),
-    url(r'^logout$', 'django.contrib.auth.views.logout', {'next_page': '/'}),
+    url(r'^logout$', views.logout_view),
```
Comment (andela-cnnadi) [inline codango/account/urls.py]: *Nitpck*: Why is this using snake case? Is this a function-based view? If it's CBV, then this should be pascal case.

File: codango/codango/settings/base.py [hunk 5]
```diff
@@ -94,6 +94,7 @@
     'middleware.activity.ActivityMiddleWare',
     'middleware.navigation.NavigationMiddleWare',
     'middleware.resource.PopularResourcesMiddleware',
+    #'middleware.logout.SessionIdleTimeout',
```
Comment (andela-cnnadi) [inline codango/codango/settings/base.py]: *Nitpick*: Get rid of this commented out code.

File: codango/account/urls.py (hunk 4 above)
Comment (NdagiStanley) [inline codango/account/urls.py]: It is a function-based view

**Pull request merged**